### PR TITLE
Ignore .meta directory

### PR DESCRIPTION
**Pull request by Insti**
It would be useful to be able to put things like, test case generators, example solutions, versioning information, changelogs etc. into the track level exercises/{slug} directories.

However all[1] the files within these directories are served to the students when they use the `exercism fetch` command.

This patch enables the usage of a `.meta` directory which will not be NOT served as part of the problem files. 

Nothing within `exercises/{slug}/.meta` (or that directory itself) will be served to the student.

See: https://github.com/exercism/x-api/issues/113 for the full discussion.

[1] There are already *some* exceptions for files with specific names, "HINTS.md" and any file containing the word "example".

### CODE REVIEW

**Comment by Insti** [comment]
I'll fix these conflicts.

**Comment by Insti** [comment]
Conflicts resolved.
@kytrinyx it would be good to get this change merged in.

It is actually a patch for an 'Exercism.io' feature rather than a trackler refactoring.

Having an ignored `.meta` directory will help us out with out re-organisation of the xruby test generators.

File: lib/trackler/file_bundle.rb [hunk 1]
```diff
@@ -1,29 +1,47 @@
 require 'zip'
 
 module Trackler
-  # FileBundle is a zippech archive of a directory.
+  # A FileBundle is a collection of files from within an exercise directory
+  # It contains all the files that will be provided by the `exercism fetch` command
+  # EXCEPT for those whose names match any of the ignore patterns.
```
Comment (kytrinyx) [inline lib/trackler/file_bundle.rb]: This is lovely. Delightfully explicit.

**Comment by kytrinyx** [comment]
This looks excellent, thank you!

**Pull request merged**